metricflow/test/table_snapshot/table_snapshots.py

```diff
@@ -0,0 +1,233 @@
+                + tuple(column_definition.type.name for column_definition in self.column_definitions)
+                + tuple(cell or "" for row in self.rows for cell in row)
+            )
+        )
+
+    @staticmethod
+    def _parse_bool_str(bool_str: str) -> bool:  # noqa: D
+        if bool_str.lower() == "false":
+            return False
+        elif bool_str.lower() == "true":
+            return True
+        else:
+            raise RuntimeError(f"Invalid string representation of a boolean: {bool_str}")
+
+    @property
+    def as_df(self) -> pd.DataFrame:  # noqa: D
+        """Return this snapshot as represented by an equivalent dataframe"""
+        # In the YAML files, all values are strings, but they need to be converted to defined type so that it can be
+        # properly represented in a dataframe
+
+        type_converted_rows = []
+        for row in self.rows:
+            type_converted_row: List[Union[str, datetime.datetime, int, float, bool, None]] = []
+            for column_num, column_value in enumerate(row):
+                column_type = self.column_definitions[column_num].type
+                if column_value is None:
+                    type_converted_row.append(None)
+                elif column_type is SqlTableColumnType.STRING:
+                    type_converted_row.append(column_value)
+                elif column_type is SqlTableColumnType.TIME:
+                    type_converted_row.append(dateutil.parser.parse(column_value))
+                elif column_type is SqlTableColumnType.INT:
+                    type_converted_row.append(int(column_value))
+                elif column_type is SqlTableColumnType.FLOAT:
+                    type_converted_row.append(float(column_value))
+                elif column_type is SqlTableColumnType.BOOLEAN:
+                    type_converted_row.append(SqlTableSnapshot._parse_bool_str(column_value))
+                else:
+                    assert_values_exhausted(column_type)
+            type_converted_rows.append(type_converted_row)
+
+        return pd.DataFrame(
+            columns=[column_definition.name for column_definition in self.column_definitions],
+            data=type_converted_rows,
+        )
+
+
+class SqlTableSnapshotRestorer:
+    """Restores a snapshot of a table into the SQL engine."""
+
+    def __init__(self, sql_client: SqlClient, schema_name: str) -> None:  # noqa: D
+        self._sql_client = sql_client
+        self._schema_name = schema_name
+
+    def restore(self, table_snapshot: SqlTableSnapshot, overwrite: bool = False) -> None:  # noqa: D
+        sql_table = SqlTable(schema_name=self._schema_name, table_name=table_snapshot.table_name)
+        if overwrite:
+            self._sql_client.drop_table(sql_table)
+        elif self._sql_client.table_exists(sql_table):
+            logger.info(f"Table {sql_table.sql} already exists, so not restoring")
```

Comment:
Maybe `already exists, no need to restore` here? Makes it more obvious that this for the case where you want to load new tables but not clobber the ones that are there.

Reply:
Updated.